packages/flamegrill/src/profile/profile.ts

```diff
@@ -20,6 +20,16 @@ export interface ScenarioProfiles {
   [scenarioName: string]: ScenarioProfile;
 };
 
+export type ProfilePage = puppeteer.Page;
+
+export interface ScenarioProfileConfig {
+  outDir: string;
+  tempDir: string;
+
+  /** Async operation which will be executed before taking metrics from page. */
+  executeBeforeMeasurement?(page: ProfilePage): Promise<void>;
```

Comment:
I purposely had this tightly coupled with `ScenarioConfig`, now it is loosely coupled. I think you can still have `executeBeforeMeasurement` work with `Required`, it would just have to be `executeBeforeMeasurement: (page: ProfilePage): Promise<void> | undefined;`